cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -571,15 +581,14 @@ class ObjectInputFile final : public io::RandomAccessFile {
   std::shared_ptr<const KeyValueMetadata> metadata_;
 };
 
-Status CreateEmptyBlockBlob(std::shared_ptr<Blobs::BlockBlobClient> block_blob_client) {
+Status CreateEmptyBlockBlob(Blobs::BlockBlobClient& block_blob_client) {
```

Comment:
If not const, can we pass it as pointer?
```suggestion
Status CreateEmptyBlockBlob(Blobs::BlockBlobClient* block_blob_client) {
```

Reply:
I can make it const. Pushing a fix.